data-prepper-plugins/aggregate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/aggregate/AggregateProcessor.java

```diff
@@ -101,10 +103,12 @@ public Collection<Record<Event>> doExecute(Collection<Record<Event>> records) {
                 handleEventsDropped++;
                 continue;
             }
+            Map<String, Object> identificationKeysMap = new HashMap<>();
+            getIdentificationKeys().stream().forEach(key -> identificationKeysMap.put(key, event.get(key, Object.class)));
             final AggregateIdentificationKeysHasher.IdentificationHash identificationKeysHash = aggregateIdentificationKeysHasher.createIdentificationKeyHashFromEvent(event);
```

Comment:
We're performing the exact same operation twice here

Reply:
Best solution is to include the `IdentificationHash` object as part of the AggregateActionInput, and to then use that to populate the groupState

Reply:
Not really. Each event will have their own set of values for the `identification_keys`. No matter how we implement, this needs to happen inside the loop.

Reply:
The call to `createIdentificationKeyHashFromEvent` is constructing the exact same map that you created with the java stream isn't it? - https://github.com/opensearch-project/data-prepper/blob/b381189e04079c6e23ac98a7f32741351862c347/data-prepper-plugins/aggregate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/aggregate/AggregateIdentificationKeysHasher.java#L22

Reply:
I am not sure why the IdentificationKeyHash needs to have the entire map. That implementation seems sub-optimal. Why is not implemented like just a hashcode (which is initialized by calling map.hash()) instead of storing the entire map and then calling the hash later?